---
title: "NIH_graphics"
format: html
editor: visual
---


```{r, message=FALSE}
knitr::opts_chunk$set(comment = NA) 
library(broom)
library(janitor) 
library(naniar)
library(glue)
library(rsample)
library(yardstick)
library(GGally)
library(mice)
library(car)
library(pROC)
library(patchwork)
library(rms)
library(simputation)
library(kableExtra)
library(tidyverse)
library(extrafont)
theme_set(theme_bw()) 
```


```{r}
font_import(paths = 'fonts', prompt = FALSE)
loadfonts()
```

```{r, fig.height=10, fig.width=18}
log_1 <- read_csv("log_VF1.csv", show_col_types = FALSE)
log_1$Feature <- gsub("apc_","",as.character(log_1$Feature))
ggplot(log_1, aes(x = Feature, y = Effect)) +
  geom_point(col = "#2b8cbe") +
  geom_hline(yintercept = 1.0, linetype = "dotted", col = "red") +
  geom_label(aes(label=round(Effect, 3)), hjust = -0.10, vjust = -0.1, size = 5) +
  geom_errorbar(aes(ymin = Lower_95, ymax = Upper_95), width = 0.2, col = "#2b8cbe", linewidth = 1.0) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))+
  labs(title = str_wrap("Predicted Effects of a change of one unit in the FAVOR aPCs on the odds of failing VFlag 1.", width = 70),
       x = "", y = "Effect",
       subtitle = "A variant fails VFlag 1 if GATK 'FILTER'!= 'PASS' or is in tranche >=99.7%. ",
       caption = "Variants taken from Chromosomes 21 and 22 from the ADSP R4 data release.") + 
  theme(text=element_text(size=22, family="Work Sans"),
        axis.text.x = element_text(angle = 60, hjust = 1, 
                                  face = ifelse(log_1$Lower_95 <= 1 & log_1$Upper_95 >= 1, "plain", "bold")))
ggsave("log_v1.png", width = 16, height = 10, dpi = 2000)

```



```{r, fig.height=10, fig.width=18, warning=FALSE}
log_2 <- read_csv("log_VF2.csv", show_col_types = FALSE)
log_2$Feature <- gsub("apc_","",as.character(log_2$Feature))
ggplot(log_2, aes(x = Feature, y = Effect)) +
  geom_point(col = "#2b8cbe") +
  geom_hline(yintercept = 1.0, linetype = "dotted", col = "red") +
  geom_label(aes(label=round(Effect, 3)), hjust = -0.10, vjust = -0.1, size = 5) +
  geom_errorbar(aes(ymin = Lower_95, ymax = Upper_95), width = 0.2, col = "#2b8cbe", linewidth = 1.0) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))+
  labs(title = str_wrap("Predicted Effects of a change of one unit in the FAVOR aPCs on the odds of failing VFlag 2.", width = 70),
       x = "", y = "Effect",
       subtitle = "A variant fails VFlag 2 if all genotypes have DP<10 and/or GQ<20. ",
       caption = "Variants taken from Chromosomes 21 and 22 from the ADSP R4 data release.") + 
  theme(text=element_text(size=22, family="Work Sans"),
        axis.text.x = element_text(angle = 60, hjust = 1, 
                                  face = ifelse(log_1$Lower_95 <= 1 & log_1$Upper_95 >= 1, "plain", "bold")))
ggsave("log_v2.png", width = 16, height = 10, dpi = 2000)

```


```{r, fig.height=14, fig.width=20}
log_3 <- read_csv("log_VF3.csv", show_col_types = FALSE)
log_3$Feature <- gsub("apc_","",as.character(log_3$Feature))
ggplot(log_3, aes(x = Feature, y = Effect)) +
  geom_point(col = "#2b8cbe") +
  geom_hline(yintercept = 1.0, linetype = "dotted", col = "red") +
  geom_label(aes(label=round(Effect, 3)), hjust = -0.10, vjust = -0.1, size = 5) +
  geom_errorbar(aes(ymin = Lower_95, ymax = Upper_95), width = 0.2, col = "#2b8cbe", linewidth = 1.0) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))+
  labs(title = str_wrap("Predicted Effects of a change of one unit in the FAVOR aPCs on the odds of failing VFlag 3.", width = 70),
       x = "", y = "Effect",
       subtitle = "A variant fails VFlag 3 if MAF = 0. ",
       caption = "Variants taken from Chromosomes 21 and 22 from the ADSP R4 data release.") + 
  theme(text=element_text(size=22, family="Work Sans"),
        axis.text.x = element_text(angle = 60, hjust = 1, 
                                  face = ifelse(log_1$Lower_95 <= 1 & log_1$Upper_95 >= 1, "plain", "bold")))
ggsave("log_v3.png", width = 16, height = 10, dpi = 2000)

```


```{r, fig.height=14, fig.width=20}
log_4 <- read_csv("log_VF4.csv", show_col_types = FALSE)
log_4$Feature <- gsub("apc_","",as.character(log_4$Feature))
ggplot(log_4, aes(x = Feature, y = Effect)) +
  geom_point(col = "#2b8cbe") +
  geom_hline(yintercept = 1.0, linetype = "dotted", col = "red") +
  geom_label(aes(label=round(Effect, 3)), hjust = -0.10, vjust = -0.1, size = 5) +
  geom_errorbar(aes(ymin = Lower_95, ymax = Upper_95), width = 0.2, col = "#2b8cbe", linewidth = 1.0) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))+
  labs(title = str_wrap("Predicted Effects of a change of one unit in the FAVOR aPCs on the odds of failing VFlag 4.", width = 70),
       x = "", y = "Effect",
       subtitle = "A variant fails VFlag 4 if Call Rate <= 80%. ",
       caption = "Variants taken from Chromosomes 21 and 22 from the ADSP R4 data release.") + 
  theme(text=element_text(size=22, family="Work Sans"),
        axis.text.x = element_text(angle = 60, hjust = 1, 
                                  face = ifelse(log_1$Lower_95 <= 1 & log_1$Upper_95 >= 1, "plain", "bold")))
ggsave("log_v4.png", width = 16, height = 10, dpi = 2000)

```



```{r, fig.height=14, fig.width=17}
log_5 <- read_csv("log_VF5.csv", show_col_types = FALSE)
log_5$Feature <- gsub("apc_","",as.character(log_5$Feature))
ggplot(log_5, aes(x = Feature, y = Effect)) +
  geom_point(col = "#2b8cbe") +
  geom_hline(yintercept = 1.0, linetype = "dotted", col = "red") +
  geom_label(aes(label=round(Effect, 3)), hjust = -0.10, vjust = -0.1, size = 4) +
  geom_errorbar(aes(ymin = Lower_95, ymax = Upper_95), width = 0.2, col = "#2b8cbe", linewidth = 1.0) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))+
  labs(title = str_wrap("Predicted Effects of a change of one unit in the FAVOR aPCs on the odds of failing VFlag 5.", width = 70),
       x = "", y = "Effect",
       subtitle = "A variant fails VFlag 5 if Average Mean Depth > 500 reads. ",
       caption = "Variants taken from Chromosomes 21 and 22 from the ADSP R4 data release.") + 
  theme(text=element_text(size=22, family="Work Sans"),
        axis.text.x = element_text(angle = 60, hjust = 1, 
                                  face = ifelse(log_1$Lower_95 <= 1 & log_1$Upper_95 >= 1, "plain", "bold")))
ggsave("log_v5.png",width = 16, height = 10, dpi = 2000)

```

